scripts/publish_docs.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+set -e
+
+### Fetch the gh-pages branch and switch to it ###
+git fetch origin gh-pages
+git checkout -qf FETCH_HEAD
+git checkout -f gh-pages || git checkout -bf gh-pages
```

Comment:
Why aren't we running with -b directly instead of first failing to checkout and then doing it?